### ISSUE TITLE
make-swf fails because flex-sdk does not install 

### ISSUE THREAD

**Issue by claytonrothschild** [google/marzipano#151]
Hi all,

flex-sdk does not get installed when I run npm install. This causes builds to fail because of missing mxlmc package found in scripts/make-swf. Anyone experience anything similar?

Steps taken:
I download the repo:

`git clone https://github.com/google/marzipano.git`

then I run npm install:

`npm install`

The installation of dependencies begins and proceeds:
![image](https://user-images.githubusercontent.com/11081953/36758738-e4094722-1bda-11e8-91c7-ae91c2ce34ea.png)

but then finishes with the following output in my console:
![image](https://user-images.githubusercontent.com/11081953/36758782-00953d74-1bdb-11e8-8337-11f6fffbf698.png)
#
It appears that flex_sdk does not install, and it is indeed missing from my node_modules:

![image](https://user-images.githubusercontent.com/11081953/36758195-5b9e3ae2-1bd9-11e8-892b-5882b5ff1d42.png)

Thus, when I run:

`npm run dev`

the build does not work:

`scripts/make-swf: line 27: mxmlc: command not found`

More verbose log of the build error:
![image](https://user-images.githubusercontent.com/11081953/36758151-289138f2-1bd9-11e8-9b41-ec5f91d90d19.png)

Your help is greatly appreciated.

**Comment by tjgq** [comment]
If there's no `flex-sdk` directory in `node_modules`, it looks like it failed to install. Can you try running `npm install --verbose flex-sdk` and look for errors?

**Comment by claytonrothschild** [comment]
Running the install command explicitly produces a similar result:


```
clayton@DevLinux:~/src/marzipano$ npm install --verbose flex-sdk
npm info it worked if it ends with ok
npm verb cli [ '/usr/bin/nodejs',
npm verb cli   '/usr/bin/npm',
npm verb cli   'install',
npm verb cli   '--verbose',
npm verb cli   'flex-sdk' ]
npm info using npm@3.10.10
npm info using node@v6.10.1
npm verb request uri https://registry.npmjs.org/flex-sdk
npm verb request no auth needed
npm info attempt registry request try #1 at 5:07:18 PM
npm verb request id 41e27f141b600e01
npm verb etag "534378aa-ae9f"
npm verb lastModified Tue, 8 Apr 2014 4:18:50 GMT
npm http request GET https://registry.npmjs.org/flex-sdk
npm http 304 https://registry.npmjs.org/flex-sdk
npm verb headers { date: 'Wed, 28 Feb 2018 17:07:20 GMT',
npm verb headers   via: '1.1 varnish',
npm verb headers   'cache-control': 'max-age=300',
npm verb headers   etag: '"534378aa-ae9f"',
npm verb headers   age: '0',
npm verb headers   connection: 'keep-alive',
npm verb headers   'x-served-by': 'cache-dfw18649-DFW',
npm verb headers   'x-cache': 'MISS',
npm verb headers   'x-cache-hits': '0',
npm verb headers   'x-timer': 'S1519837641.674326,VS0,VE68',
npm verb headers   vary: 'Accept-Encoding, Accept' }
npm verb etag https://registry.npmjs.org/flex-sdk from cache
npm verb get saving flex-sdk to /home/clayton/.npm/registry.npmjs.org/flex-sdk/.                cache.json
npm verb correctMkdir /home/clayton/.npm correctMkdir not in flight; initializin                g
npm verb cache add spec flex-sdk@^4.6.0-0
npm verb addNamed ">=4.6.0-0 <5.0.0" is a valid semver range for flex-sdk
npm verb addNameRange registry:https://registry.npmjs.org/flex-sdk not in flight                ; fetching
npm verb get https://registry.npmjs.org/flex-sdk not expired, no request
npm verb addNamed "4.6.0-0" is a plain semver version for flex-sdk
npm verb afterAdd /home/clayton/.npm/flex-sdk/4.6.0-0/package/package.json not i                n flight; writing
npm verb correctMkdir /home/clayton/.npm correctMkdir not in flight; initializin                g
npm verb afterAdd /home/clayton/.npm/flex-sdk/4.6.0-0/package/package.json writt                en
npm verb correctMkdir /home/clayton/.npm/_locks correctMkdir not in flight; init                ializing
npm verb lock using /home/clayton/.npm/_locks/staging-fc081ec33d657df6.lock for                 /home/clayton/src/marzipano/node_modules/.staging
npm verb unbuild node_modules/.staging/flex-sdk-fb3f8ad9
npm verb gentlyRm don't care about contents; nuking /home/clayton/src/marzipano/                node_modules/.staging/flex-sdk-fb3f8ad9
npm verb tar unpack /home/clayton/.npm/flex-sdk/4.6.0-0/package.tgz
npm verb tar unpacking to /home/clayton/src/marzipano/node_modules/.staging/flex                -sdk-fb3f8ad9
npm verb gentlyRm don't care about contents; nuking /home/clayton/src/marzipano/                node_modules/.staging/flex-sdk-fb3f8ad9
npm verb gentlyRm don't care about contents; nuking /home/clayton/src/marzipano/                node_modules/.staging/flex-sdk-fb3f8ad9/node_modules
npm info lifecycle flex-sdk@4.6.0-0~preinstall: flex-sdk@4.6.0-0

> flex-sdk@4.6.0-0 preinstall /home/clayton/src/marzipano/node_modules/.staging/                flex-sdk-fb3f8ad9
> npm install playerglobal-latest

```

**Comment by tjgq** [comment]
I tried installing `flex-sdk` in Ubuntu, and apparently the problem is that the install script for `playerglobal-latest` (which `flex-sdk` depends on) tries to execute `node install.js`. However, the Node interpreter name on Ubuntu is `nodejs`.

This is essentially the same issue as #73. Can you try the workaround suggested there (i.e. `sudo apt-get install nodejs-legacy`, which uses `node` instead of `nodejs` for the interpreter name)?

**Comment by claytonrothschild** [comment]
The issue does indeed seem to be related to Ubuntu as I am running Ubuntu 14.04 LTS.

Your workaround is valid and I would propose others use it.

Here are the steps I took to solve this:

I attempted to install run `sudo apt-get install nodejs-legacy` but received the following error:

```
clayton@DevLinux:~/src/marzipano$ sudo apt-get install nodejs-legacy
Reading package lists... Done
Building dependency tree
Reading state information... Done
Some packages could not be installed. This may mean that you have
requested an impossible situation or if you are using the unstable
distribution that some required packages have not yet been created
or been moved out of Incoming.
The following information may help to resolve the situation:

The following packages have unmet dependencies:
 nodejs-legacy : Depends: nodejs (>= 0.6.19~dfsg1-3~) but it is not going to be installed
E: Unable to correct problems, you have held broken packages.
```

Next, I tried to create a symlink between nodejs and node:
`ln -s /user/bin/nodejs /usr/bin/node`

but received the following error:
`ln: failed to create symbolic link ‘/usr/bin/node’: File exists`

Finally, I tried installing **Nodejs Version Manager** as proposed on this [stackoverflow.](https://stackoverflow.com/questions/21168141/cannot-install-packages-using-node-package-manager-in-ubuntu)

I was able to get a successful build by performing the following:

Install nvm:
`curl https://raw.githubusercontent.com/creationix/nvm/v0.11.1/install.sh | bash`

Run command:
`source ~/.profile`

Install 6.0.0:
`nvm install 6.0.0`

Try to build marzipano:
`npm run dev`

I then tried to view my other installed node versions to switch back to latest, but only saw 6.0.0 as being installed when I viewed available versions. 
```
clayton@DevLinux:~/src/marzipano$ nvm ls
->    v6.0.0
```
So I am unsure if my version switching fixed it (had I always been on 6.0.0?) or if perhaps the other two commands solved my problem before failing out. 

Very verbose comment here. Hopefully it helps someone or someone is able to give me clarity on which of these attempts made the build work. Will close the issue shortly if no one has any insight.

Thanks, @tjgq.

**Comment by tjgq** [comment]
My guess is that you already have an executable named `/usr/bin/node` installed by some other package, which is why you are unable to install `nodejs-legacy` or create the symlink manually. See https://lists.debian.org/debian-devel-announce/2012/07/msg00002.html.

Using nvm is also a reasonable workaround. I've added a paragraph to the README.md file to provide guidance to people facing this issue.

I don't think the Node version matters at all. AFAICT, according to the above link, the node interpreter has been called `nodejs` on Debian systems since 2012, and that is the problem here.
